kornia/color/hls.py

```diff
@@ -65,19 +63,17 @@ def rgb_to_hls(image: Tensor, eps: float = 1e-8) -> Tensor:
     l_ /= 2  # luminance
 
     # note that r,g and b were previously div by (max - min)
-    r = torch.select(im, -3, 0)
-    g = torch.select(im, -3, 1)
-    b = torch.select(im, -3, 2)
+    r, g, b = im[..., 0, :, :], im[..., 1, :, :], im[..., 2, :, :]
     # h[imax == 0] = (((g - b) / (max - min)) % 6)[imax == 0]
     # h[imax == 1] = (((b - r) / (max - min)) + 2)[imax == 1]
     # h[imax == 2] = (((r - g) / (max - min)) + 4)[imax == 2]
     cond = imax.unsqueeze(-3) == _RGB2HSL_IDX
```

Comment:
can remove the unsqueeze/squeeze too
```suggestion
    cond = imax[..., None, :, :].unsqueeze(-3) == _RGB2HSL_IDX
```